pkg/sensors/collection.go

```diff
@@ -128,6 +128,13 @@ func (c *collection) mode() tetragon.TracingPolicyMode {
 		return tetragon.TracingPolicyMode_TP_MODE_UNKNOWN
 	}
 
+	if !c.allowModeUpdate {
+		if mode != policyconf.MonitorMode {
+			logger.GetLogger().Warn("monitor-only bit is set but bpf map is not in monitor mode", "policy", c.name)
+		}
+		return tetragon.TracingPolicyMode_TP_MODE_MONITOR_ONLY
```

Comment:
comment: I wonder if we could return something else here if the mode is not monitor mode, but I don't have any good ideas. We can leave it as is and revisit as needed.